drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -31,11 +32,17 @@ LOG_MODULE_REGISTER(esp32_wifi, CONFIG_WIFI_LOG_LEVEL);
 static struct net_if *esp32_wifi_iface;
 static struct esp32_wifi_runtime esp32_data;
 
+#if defined(CONFIG_ESP32_WIFI_AP_STA_MODE)
+static struct net_if *esp32_wifi_iface_ap;
```

Comment:
I think `esp32_wifi_iface_ap` should be renamed to `esp32_wifi_iface_ap_sta` to really reflect what this instance refers to. I mean, `esp32_wifi_iface_ap` kind of forces you to consider single AP mode only.